File: back/src/main/java/com/funstat/Tables.java
package com.funstat;

import com.funstat.domain.InstrId;

public class Tables{
    public static final PersistDescriptor<InstrId> REQUESTED = new PersistDescriptor<>("requested", InstrId.class, s->s.code);
    public static final PersistDescriptor<InstrId> SYMBOLS = new PersistDescriptor<>("symbols", InstrId.class, s->s.code);
    public static final PersistDescriptor<Pair> PAIRS = new PersistDescriptor<>("pairs", Pair.class, s -> s.key);
}

File: back/src/main/java/com/funstat/Tables.kt
package com.funstat

import com.funstat.domain.InstrId

object Tables {
    val REQUESTED = PersistDescriptor("requested", InstrId::class.java) { it.code }
    val SYMBOLS = PersistDescriptor("symbols", InstrId::class.java) { it.code }
    val PAIRS = PersistDescriptor("pairs", Pair::class.java) { it.key }
}

